### PR TITLE
Competitions 

### LINKED ISSUE #3: Competitions: Add competitions section
Low fidelity prototype:

Competitions

Reference:

https://parthmittal.netlify.app/#openSource

- [ ] Use a grid to display the competitions
- [ ] It should be responsive - one card at a time on mobile, two at a time for tablets.
- [ ] The filters would be located at the top (like the reference and not like the image) for desktop. They'll be coin shaped.
- [ ] For mobile, they'll collapse and become like this (no logos, only text)  
image

- [ ] Clicking on the filter must display only that categories' competitions 
- [ ] Assets (filter logos, card images, competition details) to be provided by the Inci Media team

### PR DESCRIPTION
**Pull request by marcdhi**
## What
Made the competition cards responsive for mobile, tablet and desktop devices along with some dummy title and details.

## Why
To provide better user experience on different devices.

## How
By using CSS media queries and adjusting the layout and styles accordingly.

## To Do
- [ ] Remove hover effect from non-desktop devices
- [ ] Improve the hover effect
- [ ] Fix the card size for all devices
- [ ] Implement flip card functionality

## Issue Reference
Closes #3

### CODE REVIEW

**Comment by mittal-parth** [comment]
@marcdhi @VedantTarale Looks niiice! Few things:

- [ ] It would be nice if you could fill out the MR information clearly
- [ ] That hover effect seems off. Something more gentle would be better
- [ ] Disable that hover effect on non desktop devices
- [ ] Clicking on the card should reveal more information about the competition. Ideally you'll have just one image and not so much text on the card. Once you click on it, it will reveal text and everything in a modal (similar to the Events section). I think I missed the clicking part, so apologies for that

**Comment by marcdhi** [comment]
@mittal-parth should the flip card be the same as the events section? Or could we have different designs?

**Comment by mittal-parth** [comment]
> @mittal-parth should the flip card be the same as the events section? Or could we have different designs?

Lets stick to flipping for consistency. Later if it looks repetitive, we can think of changing it.

**Comment by mittal-parth** [comment]
By filling the MR, it means to fill out information below the specified headers and not remove them 😅
Especially write Closes `#<issue number>`. This helps automatically close the linked issue upon successful merge into main

**Comment by tarun-hegde** [comment]
@mittal-parth would  the coins size suffice?

**Comment by mittal-parth** [comment]
> @mittal-parth would the coins size suffice?

I think you can take up the entire space of the line, i.e., from the beginning of the first card to the end of the last card. Also, space them evenly instead of keeping them so close. The circles could be slightly larger as well.

I would recommend working with images rather than specifying some CSS to make it round because, ultimately that is what you'll have. 

Also, its not responsive yet, I hope you are aware.

**Comment by marcdhi** [comment]
@mittal-parth ok fixing it..

**Comment by marcdhi** [comment]
@mittal-parth should we work on the cards responsivity as well? Or are they fine?

**Comment by mittal-parth** [comment]
> @mittal-parth should we work on the cards responsivity as well? Or are they fine?

As I said, it doesn't look good for phone or tablet. It wont work out when you'll have evenly spaced elements. Rather do it like how its mentioned in the issue description 

image

Its from Devfolio, if you'd like to see how it works

File: src/components/Competitions.jsx [hunk 1]
```diff
@@ -1,9 +1,187 @@
...
+
+      {/* For coins */}
+      <div className="hidden md:block md:transition-all md:duration-100 md:flex flex-wrap md:flex-row md:space-x-9 md:justify-center md:items-center flex items-center justify-center">
+        <div class="w-auto text-white font-bold py-2 px-4 border-b-4 flex flex-row justify-evenly p-1 space-x-1">
+          {["All Events", "1", "2", "3", "4", "5", "6", "7"].map((item, index) => (
+            <button key={index}>
+              <img
+                src={logo}
+                alt=""
+                onClick={() => handleCompetitionFilter(item)}
```
Comment (BenzeneAlcohol) [inline src/components/Competitions.jsx:99]: If you click "All events", `handleCompetitionFilter` will take "All Events" as the argument, while it should be taking "All" as the argument as in L94:L101.

**Comment by mittal-parth** [comment]
image

This can be significantly improved.

- Keep a dark coloured background
- There are too many colors here, reduce
- Improve the padding on the text, card title, overall card
- You could play with font size a bit
- Have the image here as well
- Something like this is better.  
image

**Comment by marcdhi** [comment]
> image
> 
> This can be significantly improved.
> 
> * Keep a dark coloured background
> * There are too many colors here, reduce
> * Improve the padding on the text, card title, overall card
> * You could play with font size a bit
> * Have the image here as well
> * Something like this is better.
> 
> image

Ok @mittal-parth working on it

**Comment by mittal-parth** [comment]
@marcdhi @VedantTarale 

The new Modal looks great! Just small changes

- Little more padding on the heading text and below the buttons 
- The close button should be inside the modal

**Comment by marcdhi** [comment]
@mittal-parth the close button needed some animations, the existing push was just for test.. working on it 👍

**Pull request closed without merging**